### PR TITLE
Migration to jhipster v7 and and possibility to use vue.js as front

### LINKED ISSUE #301: Vue.js Support
##### **Overview of the feature request**

Vue.js Support 

##### **Motivation for or Use Case**

Vue.js has been added we can add support for this blueprint (rewrite path)
https://github.com/jhipster/generator-jhipster/pull/12064

##### **Related issues or PR**

-   [x] Checking this box is mandatory (this is just to show you read everything)

### PR DESCRIPTION
**Pull request by nicolas63**
-   Please make sure the below checklist is followed for Pull Requests.

-   [x] [All continuous integration tests](https://github.com/jhipster/generator-jhipster/actions) are green
-   [x] Tests are added where necessary
-   [x] Documentation is added/updated where necessary
-   [x] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/master/CONTRIBUTING.md) are followed

fix #301

### CODE REVIEW

**Comment by nicolas63** [comment]
@mshima i have started jhipster v7 migration for integrate vue.js on this blueprint. I was confronted with a lot of problems with the new way to manage the configuration. If you have some time can you review this PR ? 
Particularly entity/index.js I had to find a workarround since the compose phase is in the default priority, maybe you have a better solution 
I have also generation error due to prettier, there is a way to disable it  ?

**Comment by mshima** [comment]
I took a quick look.
The configuring priority should do nothing other than validating, setting a value to be saved and save it.
Currently, loading the configuration, preparing the variables for the templates and composing are done at default priority.
This should be done at the default priority: https://github.com/jhipster/jhipster-dotnetcore/blob/a432be97e9a36d5bccfb4b253fb306ae5f8fdd03/generators/client/index.js#L56-L64

Unless it depends on some input from the user, it should be as soon as possible maybe the constructor:
https://github.com/jhipster/jhipster-dotnetcore/blob/a432be97e9a36d5bccfb4b253fb306ae5f8fdd03/generators/utils.js#L100-L107

I recently created https://github.com/jhipster/generator-jhipster/issues/12338, that should improve this.
It will split current default priority into 3 (loading, preparing and default), and writing into 2 (writing, postWriting).

**Comment by nicolas63** [comment]
thank you for your feedback @mshima, the  jhipster/generator-jhipster#12338 is totally what I need

File: generators/entity/index.js [hunk 1]
```diff
@@ -57,6 +58,23 @@ module.exports = class extends EntityGenerator {
...
+                prepareEntityForTemplates(entity, this);
+
+                this.context.fields.forEach(field => {
+                    prepareFieldForTemplates(entity, field, this);
+                });
+
+                this.context.relationships.forEach(relationship => {
+                    prepareRelationshipForTemplates(entity, relationship, this);
+                });
+            },
```
Comment (mshima) [inline generators/entity/index.js:77]: Not sure what you are trying to do here.
Probably should be removed.

File: generators/entity/index.js [hunk 2]
```diff
@@ -108,6 +126,7 @@ module.exports = class extends EntityGenerator {
                         context.i18nToLoad.push(field.enumInstance);
                     }
                 });
+                this.entityConfig.fields = context.fields;
```
Comment (mshima) [inline generators/entity/index.js:129]: This saves to file. Is this required?

File: generators/entity/index.js [hunk 3]
```diff
@@ -57,6 +58,23 @@ module.exports = class extends EntityGenerator {
...
+
+                this.context.fields.forEach(field => {
+                    prepareFieldForTemplates(entity, field, this);
+                });
+
+                this.context.relationships.forEach(relationship => {
+                    prepareRelationshipForTemplates(entity, relationship, this);
+                });
+            },
             loadInMemoryDataNetBlueprint() {
```
Comment (mshima) [inline generators/entity/index.js:78]: There is a lot happening in `loadInMemoryDataNetBlueprint`.
It belongs to `preparing` priority.

Very few changes seems related to client. I would recommend to move everything possible to entity-server and entity-client for simplicity.

File: generators/server/index.js [hunk 4]
```diff
@@ -82,42 +76,25 @@ module.exports = class extends ServerGenerator {
...
-        // Here we are not overriding this phase and hence its being handled by JHipster
-        return super._default();
+        const phaseFromJHipster = super._default();
+
+        const customPhaseSteps = {
+            configureGlobalDotnetcore,
+            fixConfig() {
+                this.jhipsterConfig.serverPort = 5000;
+                this.jhipsterConfig.prodDatabaseType = 'mysql'; // set only for jdl-importer compatibility
+            },
```
Comment (mshima) [inline generators/server/index.js:94]: Save to config should happen at configuring priority.

File: generators/utils.js [hunk 5]
```diff
@@ -94,13 +94,16 @@ function configureGlobalDotnetcore() {
     this.relativeMainTestDir = `${this.relativeMainClientDir}/test`;
     this.testProjectDir = `${this.pascalizedBaseName}${constants.PROJECT_TEST_SUFFIX}`;
     this.clientTestProject = `${this.mainClientDir}/test/`;
+    this.kebabCasedBaseName = _.kebabCase(this.baseName);
```
Comment (mshima) [inline generators/utils.js:97]: configureGlobalDotnetcore seems to depend only on baseName.
baseName is supposed to be configured at app generator, so it should be settled when composing with others generators.

I recommend to rename to customizeDotnetPaths, and move to initializing priority at every generator.

**Review by mshima — COMMENTED** [review]
- Make sure that variables initialized at initializing are settled.
- Add composing, loading, preparing, postWriting priorities for completeness.
- Prefer spread notation with methods inline:
```
return {
    preTask() {},
    ...super._configuring(),
    overrideTask() {},
    postTask() {},
}
```

**Pull request merged**